src/datahike/transactor.cljc

```diff
@@ -0,0 +1,54 @@
+(ns datahike.transactor
+  (:require [datahike.core :as d]
+            [superv.async :refer [<?? <??- S thread-try]]
+            [taoensso.timbre :as log]
+            [clojure.core.async :refer [>!! chan close! promise-chan]]))
+
+(defprotocol PTransactor
+  ; Send a transaction. Returns a channel that resolves when the transaction finalizes.
+  (send-transaction! [_ tx-data tx-fn])
+  ; Returns a channel that resolves when the transactor has shut down.
+  (shutdown [_]))
+
+(defrecord LocalTransactor
+  [rx-queue rx-thread]
+  PTransactor
+  (send-transaction! [_ tx-data tx-fn]
+    (let [p (promise-chan)]
+      (>!! rx-queue {:tx-data tx-data :callback p :tx-fn tx-fn})
```

Comment:
I think it would be better to not use a thread-blocking `>!!` here, but rather `put!`. This will return immediately, but we synchronize on the returned promise anyway.

Reply:
The only problem that can arise is that there is no direct backpressure on the sender if the queue gets full.

Reply:
@whilo Yes the back pressure is exactly why I intentionally used >!! here.
If we provide a method using `put!` I think it should come with an alternative public interface which lets the client know when the put is not accepted if the buffer is full.